test/core/keyTests.ts

```diff
@@ -0,0 +1,41 @@
+﻿///<reference path="../testReference.ts" />
+
+describe("DatasetKey", () => {
+  it(" is passed to Dataset", () => {
+    let ds: Plottable.Dataset = new Plottable.Dataset();
+    let key: (d: any, i: number) => number = (d: any, i: number) => { return i; };
+    ds.keyFunction(key);
+    assert.deepEqual(ds.keyFunction(), key, "key is passed to dataset");
+  });
+  it(" may accept noConstancy predefined key function", () => {
+    let ds: Plottable.Dataset = new Plottable.Dataset();
+    ds.keyFunction(Plottable.KeyFunctions.noConstancy);
+    let d: any = { foo: "bar" };
+    let a: number = ds.keyFunction()(d, 1);
+    let b: number = ds.keyFunction()(d, 1);
+    assert.isTrue(b - a === 1, "invocations give numerically increasing results");
+  });
+  it(" defaults to useIndex predefined key function", () => {
+    let ds: Plottable.Dataset = new Plottable.Dataset();
+    assert.isTrue(ds.keyFunction() === Plottable.KeyFunctions.useIndex, "useIndex is default");
+  });
+  it(" may accept useIndex predefined key function", () => {
```

Comment:
Also, can you place a newline in between the it blocks?  Mainly to get a consistent look across the tests